gnucash/gnome-utils/gnc-gtk-utils.c

```diff
@@ -332,3 +332,320 @@ gnc_get_dialog_widget_from_id (GtkDialog *dialog, const gchar *id)
+                                            key, mods, GTK_ACCEL_VISIBLE);
+        }
+        if (GTK_IS_CONTAINER(subMenu))
+            gtk_container_foreach (GTK_CONTAINER(subMenu),
+                                   add_accel_for_menu_lookup, user_data);
+    }
+}
+
+/** Add accelerator keys for menu item widgets
+ *
+ *  @param menu The menu widget.
+ *
+ *  @param accel_group The accelerator group to use.
+ */
+void
+gnc_add_accelerator_keys_for_menu (GtkWidget *menu, GtkAccelGroup *accel_group)
+{
+    g_return_if_fail (GTK_IS_WIDGET(menu));
+    g_return_if_fail (accel_group != NULL);
+
+    gtk_container_foreach (GTK_CONTAINER(menu), add_accel_for_menu_lookup, accel_group);
+}
+
+
+static gpointer
+find_menu_item_func (GtkWidget *widget, const gchar *action_name)
+{
+    GtkWidget *ret = NULL;
+
+    if (GTK_IS_MENU_ITEM(widget))
+    {
+        const gchar *a_name = g_object_get_data (G_OBJECT(widget), "myaction-name");
+
+        GtkWidget* subMenu;
+
+        if (g_strcmp0 (a_name, action_name) == 0)
+            return widget;
+
+        subMenu = gtk_menu_item_get_submenu (GTK_MENU_ITEM(widget));
+
+        if (GTK_IS_CONTAINER(subMenu))
+        {
+            GList *container_list = gtk_container_get_children (GTK_CONTAINER(subMenu));
+            for (GList *n = container_list; !ret && n; n = n->next)
+                ret = find_menu_item_func (n->data, action_name);
+            g_list_free (container_list);
+        }
+    }
+    return ret;
+}
+
+/** Search the menu for the menu item based on the label or action name
+ *
+ *  @param menu The menu widget.
+ *
+ *  @param action_name The GAction name.
+ *
+ *  @return The menu item widget or NULL.
+ */
+GtkWidget *
+gnc_find_menu_item (GtkWidget *menu, const gchar *action_name)
+{
+    GtkWidget *ret = NULL;
+
+    g_return_val_if_fail (GTK_IS_WIDGET(menu), NULL);
+    g_return_val_if_fail (action_name != NULL, NULL);
+
+    if (GTK_IS_CONTAINER(menu))
+    {
+        GList *container_list = gtk_container_get_children (GTK_CONTAINER(menu));
+        for (GList *n = container_list; !ret && n; n = n->next)
+            ret = find_menu_item_func (n->data, action_name);
+        g_list_free (container_list);
+    }
+    return ret;
+}
+
+
+struct menu_list
+{
+    GList *list;
+};
+
+static void
+search_menu_item_list (GtkWidget *widget, gpointer user_data)
+{
+    struct menu_list *ml = user_data;
+
+    if (GTK_IS_MENU_ITEM(widget))
+    {
+        GtkWidget* subMenu = gtk_menu_item_get_submenu (GTK_MENU_ITEM(widget));
+        const gchar *a_name = g_object_get_data (G_OBJECT(widget), "myaction-name");
+
+        ml->list = g_list_prepend (ml->list, widget);
+
+        if (!a_name)
+        {
+            GtkWidget *accel_label = gtk_bin_get_child (GTK_BIN(widget));
+
+            if (accel_label)
+            {
+                // use gtk_label_get_text to get text with no underlines
+                const gchar *al_name = gtk_label_get_label (GTK_LABEL(accel_label));
```

Comment:
https://gist.github.com/christopherlam/565fc8d2fbd0511662ae6f7c14f7a30f

Reply:
@christopherlam When I made my original change I wanted to do as you have done but just could not get the correct syntax, I can see from your gist where I was going wrong, thank you.